crates/neon/src/macro_internal/mod.rs

```diff
@@ -17,46 +28,86 @@ pub static EXPORTS: [for<'cx> fn(&mut ModuleContext<'cx>) -> NeonResult<Export<'
 #[linkme::distributed_slice]
 pub static MAIN: [for<'cx> fn(ModuleContext<'cx>) -> NeonResult<()>];
 
-// Provides an identically named method to `NeonExportReturnJson` for easy swapping in macros
-pub trait NeonExportReturnValue<'cx> {
-    fn try_neon_export_return(self, cx: &mut Cx<'cx>) -> JsResult<'cx, JsValue>;
+// Wrapper for the value type and return type tags
+pub struct NeonMarker<Tag, Return>(PhantomData<Tag>, PhantomData<Return>);
```

Comment:
Possible renaming to clarify roles of these markers.

```rust
pub struct NeonMarker<Transform, ReturnType>(PhantomData<Transform>, PhantomData<ReturnType>);

pub struct NeonJsonTransform;
pub struct NeonIdentityTransform;
pub struct NeonResultReturn;
pub struct NeonValueReturn;
```

Does this improve readability?